x/evm/types/msg.go

```diff
@@ -153,6 +154,30 @@ func (msg MsgEthereumTx) Route() string { return RouterKey }
 // Type returns the type value of an MsgEthereumTx.
 func (msg MsgEthereumTx) Type() string { return TypeMsgEthereumTx }
 
+func ValidateChainID(tx *ethtypes.Transaction) error {
+	if tx.ChainId().Sign() == 0 {
+		return errorsmod.Wrap(errortypes.ErrInvalidChainID, "chain ID must be present on AccessList txs")
+	}
+	return nil
+}
+
+func ValidateDynamicFeeTx(tx *ethtypes.Transaction) error {
+	gasPrice := tx.GasTipCap()
+	if tx.GasFeeCapIntCmp(gasPrice) < 0 {
+		return errorsmod.Wrapf(
+			ErrInvalidGasCap,
+			"max priority fee per gas higher than max fee per gas (%s > %s)",
+			gasPrice, tx.GasFeeCap(),
+		)
+	}
+	gasLimit := new(big.Int).SetUint64(tx.Gas())
+	fee := new(big.Int).Mul(gasPrice, gasLimit)
+	if !types.IsValidInt256(fee) {
```

Comment:
go-ethereum doesn't validate this, it'll end up with an "insurfficient balance" anyway.